doc/faq/osx_framework.rst

```diff
@@ -0,0 +1,129 @@
+.. _osxframework-faq:
+
+******************************
+Working with Matplotlib on OSX
+******************************
+
+.. contents::
+   :backlinks: none
+
+
+.. _osxframework_introduction:
+
+Introduction
+============
+
+On OSX, two different types of Python Builds exist: a regular build and a
+framework build. In order to interact correctly with OSX through the native
+GUI frameworks you need a framework build of Python.
+At the time of writing the ``macosx`` and ``WXAgg`` backends require a
+framework build to function correctly. This can result in issues for
+a python installation not build as a framework and may also happen in 
+virtual envs and when using (Ana)Conda.
+From Matplotlib 1.5 onwards the ``macosx`` backend
+checks that a framework build is available and fails if a non framework
+build is found. WX has a similar check build in.
+
+Without this check a partially functional figure is created.
+Among the issues with it is that it is produced in the background and
+cannot be put in front of any other window. Several solutions and work 
+arounds exist see below.
+
+Short version
+=============
+
+VirtualEnv
+----------
+
+If you are on Python 3, use 
+`venv <https://docs.python.org/3/library/venv.html>`_
+instead of `virtualenv <https://virtualenv.pypa.io/en/latest/>`_::
+
+    python -m venv my-virtualenv
+    source my-virtualenv/bin/activate
+
+Otherwise you will need one of the workarounds below.
+
+Conda
+-----
+
+The default python provided in (Ana)Conda is not a framework
+build. However, the Conda developers have made it easy to install
+a framework build in both the main environment and in Conda envs.
+To use this install python.app ``conda install python.app`` and
+use ``pythonw`` rather than ``python``
+
+
+Long version
+============
+
+Unfortunately virtualenv creates a non
```

Comment:
Worth another mention of `venv` here in case someone skipped straight to the long version?